handling-forms/controlled-forms.md

```diff
@@ -0,0 +1,542 @@
+
+Let's see how this all plays out by creating a small example.
+
+Let's create a component named `CityNameInput`. This component has a text input field, where users will be able to name a city. The city name will be displayed, too.
+
+<!-- prettier-ignore-start -->
+```js
+import { useState } from 'react';
+
+const CityNameInput = () => {
+  const [cityName, setCityName] = useState('Seattle');
+
+  return (<section>
+    <h2>{cityName}</h2>
+    <input type="text"></input>
+  </section>)
+}
+```
+<!-- prettier-ignore-end -->
+
+In order to complete this form, we will do the following:
+
+1. Ensure that our text input field's value comes from the component's state
+1. Ensure that when our text input field changes, it updates the component's state
+
+We set the text input field's value by using the `value` attribute:
+
+<!-- prettier-ignore-start -->
+```js
+import { useState } from 'react';
+
+const CityNameInput = () => {
+  const [cityName, setCityName] = useState('Seattle');
+
+  return (<section>
+    <h2>{cityName}</h2>
+    <input type="text" value={cityName}></input>
+  </section>)
+}
+```
+<!-- prettier-ignore-end -->
+
+Because component's re-render every time state updates, we can trust that the appearance of the text field will always be aligned with state.
+
+Next, we need to create an event handler for the "change" event that updates state.
+
+<!-- prettier-ignore-start -->
+```js
+import { useState } from 'react';
+
+const CityNameInput = () => {
+  const [cityName, setCityName] = useState('Seattle');
+
+  const renameCity = (changeEvent) => {
+    console.log('Details about the element that fired the event:', changeEvent.target);
+    console.log('The value of that element:', changeEvent.target.value);
+    setCityName(changeEvent.target.value);
+  }
+
+  return (<section>
+    <h2>{cityName}</h2>
+    <input type="text" value={cityName} onChange={renameCity}></input>
+  </section>)
+}
+```
+<!-- prettier-ignore-end -->
+
+In this example, we added the attribute `onChange` to our text field. Then, we created an event handler function named `renameCity`.
+
+`renameCity` took in one argument, `changeEvent`. Every event-handling function will automatically be passed [an Event object](https://developer.mozilla.org/en-US/docs/Web/API/Event).
+
+Event objects include all sorts of details about the event, and we can look through [the Event documentation](https://developer.mozilla.org/en-US/docs/Web/API/Event/target) to find anything we need.
+
+Our goal is to update our `cityName` state to whatever the value of the text field is. To achieve this, we use our update function, `setCityName`, and pass in `changeEvent.target.value`.
+
+### Verifying
+
+When we run this app, we see our text field and component state all aligned!
+
+![](../assets/handling-forms_controlled-forms_renamecity.png)
+
+The most important thing about this code is that the component's state changed. We can even use our React Dev Tools to check the component's state.
```

Comment:
we change lots of state in other examples, so the way this is phrased seems a little over the top. included more explicit text about what the code does
```suggestion
The most important thing about this code is that we used the data from the event to update our component's state. We can confirm this happened using our React Dev Tools in the Components tab.
```